### PR TITLE
[MIG] hr_transfer: migrate to 14.0

### PR DESCRIPTION
**Pull request by al3ba**

### CODE REVIEW

File: hr_transfer/models/hr_transfer.py [hunk 1]
```diff
@@ -0,0 +1,301 @@
...
+# Copyright (C) 2013 Michael Telahun Makonnen <mmakonnen@gmail.com>.
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html).
+
+from dateutil.relativedelta import relativedelta
+
+from odoo import _, api, fields, models
+from odoo.exceptions import AccessError, UserError
+
+
+class HrTransfer(models.Model):
```
Comment (mtelahun) [inline hr_transfer/models/hr_transfer.py:11]: Multi-company consistency:
_check_company_auto needs to be set
check_company=True needs to be set on relational fields whose models contain a company_id

https://www.odoo.com/documentation/15.0/developer/howtos/company.html

File: hr_transfer/models/hr_transfer.py [hunk 2]
```diff
@@ -0,0 +1,301 @@
...
+            ("draft", "Draft"),
+            ("confirm", "Confirmed"),
+            ("pending", "Pending"),
+            ("done", "Done"),
+            ("cancel", "Cancelled"),
+        ],
+        default="draft",
+        readonly=True,
+    )
+    company_id = fields.Many2one(
```
Comment (mtelahun) [inline hr_transfer/models/hr_transfer.py:90]: Maybe this needs to stop being a related field, and instead should just have a default value. Not sure which is the right way to go.

File: hr_transfer/models/hr_transfer.py [hunk 3]
```diff
@@ -0,0 +1,301 @@
...
+        comodel_name="res.company",
+        related="employee_id.company_id",
+        store=True,
+        readonly=True,
+    )
+
+    @api.depends("employee_id")
+    def _compute_onchange_employee(self):
+
+        for transfer in self.filtered("employee_id"):
```
Comment (mtelahun) [inline hr_transfer/models/hr_transfer.py]: I'm not sure filtered() does what you think it does in this case.

File: hr_transfer/security/ir.model.access.csv [hunk 4]
```diff
@@ -0,0 +1,4 @@
+id,name,model_id:id,group_id:id,perm_read,perm_write,perm_create,perm_unlink
+access_hr_transfer_default,access_hr_transfer_default,model_hr_department_transfer,,0,0,0,0
```
Comment (mtelahun) [inline hr_transfer/security/ir.model.access.csv]: What use is this?

File: hr_transfer/data/hr_transfer_cron.xml [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+        <field name="interval_type">days</field>
+        <field name="numbercall">-1</field>
+        <field
+            name="nextcall"
+            eval="(DateTime.now() + timedelta(hours= +(24-DateTime.now().hour))).strftime('%Y-%m-%d %H:%M:%S')"
+        />
+        <field name="doall" eval="True" />
+        <field name="model_id" ref="model_hr_department_transfer" />
+        <field name="code">model.try_pending_department_transfers()</field>
+    </record>
```
Comment (mtelahun) [inline hr_transfer/data/hr_transfer_cron.xml]: The xml data to declare scheduled actions has changed in v14.

**Review by mtelahun — CHANGES_REQUESTED** [review]
Ah sorry. I didn't finish the review.

File: hr_transfer/data/hr_transfer_cron.xml (hunk 5 above)
Comment (al3ba) [reply hr_transfer/data/hr_transfer_cron.xml]: I have made the changes introduced in v11 (model -> model_id and function + args -> code) apart from that, I can't find any changes introduced in v14.

File: hr_transfer/security/ir.model.access.csv (hunk 4 above)
Comment (al3ba) [reply hr_transfer/security/ir.model.access.csv]: sorry, it was a typo

File: hr_transfer/security/ir.model.access.csv [hunk 6]
```diff
@@ -0,0 +1,4 @@
+id,name,model_id:id,group_id:id,perm_read,perm_write,perm_create,perm_unlink
+access_hr_transfer_default,access_hr_transfer_default,model_hr_department_transfer,,1,0,0,0
```
Comment (mtelahun) [inline hr_transfer/security/ir.model.access.csv]: should the group be empty? or should it be base.group_user

**Review by mtelahun — APPROVED** [review]
I only have one additional comment on the group access csv. Other than that looks good. If you can address my comment go ahead and merge.

**Pull request merged**